src/Angular.js

```diff
@@ -1772,6 +1772,48 @@ function getTestability(rootElement) {
   return injector.get('$$testability');
 }
 
+function rootElementProviderFactory(rootElement) {
+  return ['$shutdownProvider', function($shutdownProvider) {
+    $shutdownProvider.register(function() {
+      if (rootElement.dealoc) {
+        rootElement.dealoc();
+      } else {
+        rootElement.find('*').removeData();
```

Comment:
Maybe something like this (which is not super efficient, but shouldn't really matter, when shutting down):

```js
var unvisitedNodes = [$rootElement[0]];
var foundNodes = [$rootElement[0]];
var node;

while ((node = unvisitedNodes.pop())) {
  var newNodes = Array.prototype.filter.call(node.children, function (child) {
    return directiveNormalize(nodeName_(child)) !== 'ngNonBindable';
  });
  Array.prototype.push.apply(unvisitedNodes, newNodes);
  Array.prototype.push.apply(foundNodes, newNodes);
}

jqLite.cleanData(foundNodes);
```